pkg/util/httpRetriever.go

```diff
@@ -45,24 +49,33 @@ func _recReq(req *HTTPFetchRequest, expiration time.Time) ([]byte, error) {
 	}
 	if err != nil {
 		// Log local non-timeout errors for now.
-		httpFetchLog.Warn("Problem fetching data from: %s. %v", req.QueryURL, err)
+		level.Warn(logger).Log(
+			"msg", "problem fetching data",
+			"queryURL", req.QueryURL,
+			"err", err,
+		)
 		now := time.Now()
 		if now.After(expiration) {
-			httpFetchLog.Error("Timeout expired, not retrying query and passing error up")
+			level.Error(logger).Log("msg", "timeout expired, not retrying query and passing error up")
 			return nil, err
```

Comment:
In places like this where we return the error there is no need to log it as the caller will log it anyway. Instead should just wrap it.
```suggestion
			return nil, errors.Wrap(err,"timeout expired, not retrying query")
```